File: backend/app/schemas/artist.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class ArtistBase(BaseModel):
    artist_name: str
    bio: Optional[str] = None
    image_url: Optional[str] = None


class ArtistCreate(ArtistBase):
    pass


class ArtistUpdate(BaseModel):
    artist_name: Optional[str] = None
    bio: Optional[str] = None
    image_url: Optional[str] = None


class ArtistRead(ArtistBase):
    artist_id: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class EventArtistCreate(BaseModel):
    event_day_id: int
    artist_id: int
    is_backup: bool = False


class EventArtistRead(BaseModel):
    event_artist_id: int
    event_day_id: int
    artist_id: int
    artist_name: str
    is_backup: bool

    model_config = {"from_attributes": True}

File: frontend/src/types/artist.ts
export type Artist = {
  artist_id: number;
  artist_name: string;
  bio?: string | null;
  image_url?: string | null;
  created_at: string;
  updated_at: string;
};

export type ArtistCreate = {
  artist_name: string;
  bio?: string;
  image_url?: string;
};

export type EventArtistCreate = {
  event_day_id: number;
  artist_id: number;
  is_backup: boolean;
};

export type EventArtistRead = EventArtistCreate & {
  event_artist_id: number;
  artist_name: string;
};

